### PR TITLE
Change lemming's g* services to run on different ports per intent.

### PR DESCRIPTION
**Pull request by wenovus**
- gNMI/gNOI/gNSI on one port
- gRIBI on another
- P4RT on another

These were intended to run on different ports.

NOTE: There is actually a hack in here where I made the gRIBI service actually register in the same place as gNMI/gNOI. This is to deal with the fact that KNE doesn't have lemming registered with the right port. This can be changed after that is done.

### CODE REVIEW

File: lemming.go [hunk 1]
```diff
@@ -87,6 +106,60 @@ func WithDeviceConfig(c []byte) Option {
...
+
+// WithP4RTAddr is a device option that specifies that the P4RT address.
+func WithP4RTAddr(host string, port int) Option {
+	return func(o *opt) {
+		o.p4rtAddr.host = host
+		o.p4rtAddr.port = port
+	}
+}
+
+// gNMIAddress is the internal implementation that specifies the port that gNMI should
```
Comment (DanG100) [inline lemming.go:133]: all this can be removed?

File: lemming.go [hunk 2]
```diff
@@ -87,6 +106,60 @@ func WithDeviceConfig(c []byte) Option {
 	}
 }
 
+// WithGRIBIAddr is a device option that specifies that the gRIBI address.
+func WithGRIBIAddr(host string, port int) Option {
```
Comment (DanG100) [inline lemming.go:110]: it seems like we could just be using a string everywhere, there no reason to separate host and port, unless i'm missing something

File: lemming.go (hunk 1 above)
Comment (wenovus) [reply lemming.go]: Done

File: lemming.go (hunk 2 above)
Comment (wenovus) [reply lemming.go]: Done

**Comment by wenovus** [comment]
Note I created a PR in KNE to change the ports -- recompiling kne_cli alone wasn't enough to fix this issue, need your help here @DanG100 : https://github.com/openconfig/kne/pull/318

**Review by DanG100 — APPROVED** [review]

**Pull request merged**